src/software/thunderscope/gl/layers/gl_referee_info_layer.py

```diff
@@ -0,0 +1,269 @@
+import math
+import time
+from typing import Optional
+
+from PyQt6 import QtGui
+from google.protobuf.json_format import MessageToDict
+from pyqtgraph.opengl.items.GLTextItem import GLTextItem
+
+from proto.import_all_protos import *
+from software.py_constants import *
+from software.thunderscope.constants import DepthValues, Colors
+from software.thunderscope.gl.graphics.gl_circle import GLCircle
+from software.thunderscope.gl.graphics.gl_label import GLLabel
+from software.thunderscope.gl.helpers.observable_list import ObservableList
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+
+class GLRefereeInfoLayer(GLLayer):
+    """GLLayer that visualizes referee info"""
+
+    REFEREE_COMMAND_PREFIX = "Command: "
+    GAMESTATE_PREFIX = "Game State: "
+
+    # outline color of the avoid area for ball placement
+    BALL_PLACEMENT_ROBOT_AVOID_AREA_VISUALIZATION_COLOR = Colors.RED
+    # outline color of the tolerance region for ball placement (when the ball is inside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_ON = Colors.GREEN
+    # outline color of the tolerance region for ball placement (when the ball is outside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_OFF = Colors.RED
+    # outline color of the target mark
+    BALL_PLACEMENT_TARGET_VISUALIZATION_COLOR = Colors.FIELD_LINE_COLOR
+    # text color for count down
+    COUNT_DOWN_TEXT_COLOR = Colors.RED
+
+    @staticmethod
+    def is_point_in_circle(
```

Comment:
this helper isn't actually necessary, we've binded this function in python from C++

[C++ impl](https://github.com/UBC-Thunderbots/Software/blob/10bded3efa3780478ae10a76840a5edf17abb508/src/software/geom/algorithms/contains.h#L19)
[Python bindings](https://github.com/UBC-Thunderbots/Software/blob/10bded3efa3780478ae10a76840a5edf17abb508/src/software/python_bindings.cpp#L313)

You can call this via `tbots_cpp.contains(circle, point)`

Reply:
Done